### ISSUE TITLE
[Enhancement]: Bring windows back with python-milvus-server project.

### ISSUE THREAD

**Issue by matrixji** [milvus-io/milvus#21309]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

Days before, I was staring a project: [python-milvus-server](https://github.com/matrixji/python-milvus-server), currently it could simply run a test/develop milvus instance in seconds, and no docker env is required. (works on windows/Linux/macOS-arm64).

Currently, this project uses the code from my forked branch, I want later could release it with an official tag/release. So I hope the adaptation code can be merged upstream.

### Why is this needed?

I see some people want embd-milvus under windows, I've tried, and I see there are still some problems with embed's solution:
- No stop milvus provided, actually, it's difficult to implement it for stop milvus thread/goroutine without exit python process.
- There is still requires user/developer to set some env for starting milvus.

### Anything else?

I'll provide the adaptation code for branch 2.2.0 and master.

**Comment by xiaofan-luan** [comment]
> ant later could release it with an official tag/release. So I hope the adaptation code can be merged upstream.

what changes does it needed?
We used to support window but it's a little bit too complicated because we didn't add windows ci so it's very easy to break window compilation

**Comment by matrixji** [comment]
> > ant later could release it with an official tag/release. So I hope the adaptation code can be merged upstream.
> 
> what changes does it needed? We used to support window but it's a little bit too complicated because we didn't add windows ci so it's very easy to break window compilation

**For build under windows**
- Currently, small changes are needed on the 2.2 branch, which is already in PR: #21310.
- For the master, the compilation of the C++ part is moving to Conan, currently, my idea is to use a dummy(empty) Conan dependency for compilation under MINGW/MSYS.  Which seems could reuse most of the makefile and scripts from *nix based solution. I'm integrating with the below commit: https://github.com/matrixji/milvus/commit/7bd94353f862c9fc22dd7097e815c987bf5090c8 , the PR for the master will be based on this.
- I hope in the CI pipeline, at least we should ensure compilation succeeds under Windows.

**Raw idea about python-milvus-server**
I hope users like data scientists or AI researchers could simply use milvus in the python environment e.g. Jupiter notebook.
Firstly, I was trying to use embd-milvus on windows, and then I found some limitations in embd-milvus itself, so I decided to start a sub-process-based milvus instance with python (this project is trying to target the mainstream desktop platforms, not only windows)
